### PR TITLE
perf(trajectory_follower_nodes): change longitudinal control to use period parameter

### PR DESCRIPTION
**Pull request by Sharrrrk**
## Description

To solve issue #701 

## Tests performed

On vehicle test is performed to check the control stability.
1. Plan a path
2. Engage vehicle
3. Check the speed and acceleration control stability.

## Notes for reviewers

`tier4_control_launch` package is also updated accordingly, and duplicated parameters removed the `launch.py`.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by Sharrrrk** [comment]
@TakaHoribe would you please help me to review this PR?

**Comment by TakaHoribe** [comment]
@Sharrrrk Sure. Thank you for your great contribution!

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ AutonomouStuff Lexus RX 450h for under 40 km/h driving.
 
 | Name                                 | Type   | Description                                                                                                                                                                             | Default value |
 | :----------------------------------- | :----- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------ |
-| control_rate                         | double | control rate [Hz]                                                                                                                                                                       | 30            |
+| longitudinal_ctrl_period             | double | longitudinal control period [s]                                                                                                                                                         | 0.03          |
```
Comment (maxime-clem) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:151]: There is a 10% change between a control rate of `30Hz` and a period of `0.03s` (~`33Hz`).
I think it is better to put the period to a value of `0.0333s` to reduce the change to 0.1%.

**Review by maxime-clem — COMMENTED** [review]
Tested in the planning simulator and everything works fine.
Code changes look good and I only have a few minor comments.

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md (hunk 1 above)
Comment (maxime-clem) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:151]: A period of 0.03s corresponds to a period of ~33Hz which is 10% higher than the previous value.
It probably does not make a big difference but I think it is better to put a period of `0.0333` to minimize the difference with the previous default value.
What do you think ?

File: launch/tier4_control_launch/launch/control.launch.py [hunk 2]
```diff
@@ -93,9 +93,7 @@ def launch_setup(context, *args, **kwargs):
             lon_controller_param,
             vehicle_info_param,
             {
-                "control_rate": LaunchConfiguration("control_rate"),
```
[the remark this answers is not in the document]
Comment (Sharrrrk) [reply launch/tier4_control_launch/launch/control.launch.py:96]: This is a duplicated parameter which is already defefined in the parameter file. So I removed it. It's already included in the `lon_controller_param` file which is `launch/tier4_control_launch/config/trajectory_follower/longitudinal_controller.param.yaml` in this case. Same reason for parameter `enable_smooth_stop`.
I remove the hard coded definition in launch file, and uses the parameter in config fie.

File: launch/tier4_control_launch/launch/control.launch.py [hunk 3]
```diff
@@ -93,9 +93,7 @@ def launch_setup(context, *args, **kwargs):
...
                 "show_debug_info": LaunchConfiguration("show_debug_info"),
-                "enable_smooth_stop": LaunchConfiguration("enable_smooth_stop"),
```
[the remark this answers is not in the document]
Comment (Sharrrrk) [reply launch/tier4_control_launch/launch/control.launch.py:98]: Same as above.

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md (hunk 1 above)
Comment (Sharrrrk) [reply control/trajectory_follower_nodes/design/longitudinal_controller-design.md:151]: The reason I did not change period to 0.0333 is mentioned in #701 . The original parameter of 33.33333 ms in period is not a possible period for real vehicle control interface. So I chose 30ms instead which has the minimum gap and also consistent with real vehicle interfaces.

**Review by maxime-clem — APPROVED** [review]

**Pull request merged**